src/Core/RulesAccessTrait.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\rules\Core\RulesAccessTrait.
+ */
+
+namespace Drupal\rules\Core;
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Session\AccountInterface;
+
+/**
+ * Implements access related functions for Rules plugins.
+ */
+trait RulesAccessTrait {
```

Comment:
Let's not talk about Rules here ,this is a generic class. I'd suggest "PluginManagerAccessTrait" - the namespace is enough in this case imho. We need to provide an interface for that also though, e.g. PluginManagerAccessInterface